src/devices/sonoff.ts

```diff
@@ -433,7 +433,11 @@ const definitions: Definition[] = [
                 .withLocalTemperature()
                 .withLocalTemperatureCalibration(-7.0, 7.0, 0.2)
                 .withSystemMode(['off', 'auto', 'heat'], ea.ALL, 'Mode of the thermostat')
-                .withRunningState(['idle', 'heat'], ea.STATE_GET),
+                .withRunningState(['idle', 'heat'], ea.STATE_GET)
+                // TRVZB does not appear to support the Get Weekly Schedule command (0x02).
```

Comment:
Do you get an error when trying to get the schedule?

Reply:
Oh, I didn't realise this failed the tests. 

I've attached a trace showing the ZCL packets when pairing the TRV with Z2M. 

Packet 46 is Z2M sending the Get Weekly Schedule command
Packet 47 is the TRV responding with Unsupported Cluster Command (0x81) - which is why I removed the GET access.

Also notice how packets 35 to 41 show the TRV sending each day's schedule as a separate command - from what I can tell, this happens after pairing and after sending a Set Weekly Schedule command to the device. I haven't yet found a way to explicitly ask for the schedule, given the official command fails. The device supports the ability to configure different schedules for each day, which is why I guess it reports each day's schedule in a single packet. My understanding of Z2M is that this is currently not supported - from what I can tell, you can only configure a single schedule in Z2M and apply it to multiple days, and not have different schedules for different days. (I could be wrong though!) As a result, when Z2M receives these messages, the last message "wins":

Current state of the device in Z2M:

```
{
    "battery": 38,
    "battery_low": null,
    "child_lock": "UNLOCK",
    "frost_protection_temperature": 7,
    "linkquality": 184,
    "local_temperature": 16.5,
    "local_temperature_calibration": 0,
    "open_window": "ON",
    "running_state": "heat",
    "occupied_heating_setpoint": 19,
    "system_mode": "heat",
    "weekly_schedule": {
        "days": [
            "sunday"
        ],
        "transitions": [
            {
                "heating_setpoint": 5,
                "time": 0
            },
            {
                "heating_setpoint": 30,
                "time": 878
            },
            {
                "heating_setpoint": 5,
                "time": 879
            },
            {
                "heating_setpoint": 5,
                "time": 879
            },
            {
                "heating_setpoint": 5,
                "time": 879
            },
            {
                "heating_setpoint": 5,
                "time": 879
            }
        ]
    }
}
```

The `weekly_schedule` in the state above reflects the payload received in packet 41; the schedules in packets 35 to 40 were overwritten by packet 41.


[Sonoff Weekly Schedule.pcapng.zip](https://github.com/Koenkk/zigbee-herdsman-converters/files/13325265/Sonoff.Weekly.Schedule.pcapng.zip)

Reply:
[Sonoff Z2M Set Schedule.pcapng.zip](https://github.com/Koenkk/zigbee-herdsman-converters/files/13325307/Sonoff.Z2M.Set.Schedule.pcapng.zip)

Here's another little trace showing the current behaviour of when setting the weekly schedule from Z2M.

Packet 1 is a Set Weekly Schedule command from Z2M to the device for a schedule on Wednesday.
Packet 4 is the device responding with a Get Weekly Schedule Response command with Wednesday's schedule. (It only sends back a command for the days affected by the previous Set Weekly Schedule command).

Packet 5 is another Set Weekly Schedule command - I added Saturday to the list of days in Z2M.
Packets 8 and 9 are the responses from the TRV as they were the days affected by the previous command.

It feels like if we want to be able to fully support this, then Z2M might need to cater for the ability to configure separate schedules for each day.

Reply:
If I understand correctly, when setting the schedule multiple times, you can configure different schedules per day currently? I know it will not be reflected in `weekly_schedule` but this is just a matter of updating https://github.com/Koenkk/zigbee-herdsman-converters/blob/c3e653bf4a221b121afba26f41f4702bd071b298/src/converters/toZigbee.js#L1226

Reply:
I've added a comment on the original PR to see whether the author considered the idea. I do not see any current way of setting different schedules per day in the frontend.

https://github.com/Koenkk/zigbee-herdsman-converters/pull/5550#issuecomment-1809814572

Reply:
@pawol pointed out that this Sonoff device is very similar to, if not a rebrand of, the Tuya TS0601 _TZE200_z1tyspqw TRV. It looks like this device supports different schedules for each day and has been implemented by[ exposing a text box for each day with multiple time/temperature strings](https://github.com/Koenkk/zigbee-herdsman-converters/blob/9ad8015a141c9afd38c75f7093bd14cac5441570/src/devices/tuya.ts#L3169).

If you're happy with this, I can try to replicate something similar for this device.

Reply:
@photomoose the TS0601 uses the TuYa datapoints, while this device uses the standard ZCL Zigbee commands, so maybe it looks the same, but the software is definitely different.

Reply:
@Koenkk sure, I understand the devices are different. What I meant was I could write something similar to the Tuya device - i.e. expose 7 textboxes in the UI (one for each day) and parse the text as a series of times and temperatures.

For example:

Monday: "09:00/19 12:00/16 18:00/20"
Tuesday: "09:00/19 12:00/16 18:00/20"
...
Sunday: "09:00/19 12:00/16 18:00/20"

I think this might be easier to do than trying to make the current front end support this schedule format.

Reply:
Agree!